pkg/kubelet/dockershim/network/hostport/hostport_syncer.go

```diff
@@ -122,7 +123,7 @@ func getPodFullName(pod *PodPortMapping) string {
 func gatherAllHostports(activePodPortMappings []*PodPortMapping) (map[*PortMapping]targetPod, error) {
 	podHostportMap := make(map[*PortMapping]targetPod)
 	for _, pm := range activePodPortMappings {
-		if pm.IP.To4() == nil {
+		if pm.IP.To4() == nil && pm.IP.To16() == nil {
```

Comment:
isn't this the same as just:
```suggestion
		if pm.IP.To16() == nil {
```

Reply:
yeah, didn't know that IP.To16() works for IPv4 and IPv6 addresses.
It also applies to https://github.com/kubernetes/kubernetes/pull/80854/files#diff-b0ee819c32e4070b3a2cb6376b72408bR86

Reply:
my concern is if this will be clear for another person reading the code, the current `if statement` makes clear that checks that the IP address is neither IPv4 nor IPV6, using only To16() can lead to confusion IMHO

Reply:
I disagree, unless they are unclear on what To16() is. However, if you wanted to be most clear I'd wrap this in a method like `isValid(ip)`

Reply:
done on both files